notes.go

```diff
@@ -0,0 +1,97 @@
+package chartmogul
+
+// Note is the customer note as represented in the API.
+type Note struct {
+	UUID string `json:"uuid"`
+	// Basic info
+	CustomerUUID string `json:"customer_uuid"`
+	Type         string `json:"type"`
+	Text         string `json:"text"`
+	Author       string `json:"author"`
+	CallDuration uint32 `json:"call_duration"`
+	CreatedAt    string `json:"created_at"`
+	UpdatedAt    string `json:"updated_at"`
+}
+
+// UpdateNote allows updating note on the update endpoint.
+type UpdateNote struct {
+	Text         string `json:"text"`
+	AuthorEmail  string `json:"author_email"`
+	CallDuration uint32 `json:"call_duration"`
+	CreatedAt    string `json:"create_at"`
+	UpdatedAt    string `json:"updated_at"`
+}
+
+// NewNote allows creating note on a new endpoint.
+type NewNote struct {
+	// Obligatory
+	CustomerUUID string `json:"customer_uuid"`
+	Type         string `json:"type"`
+
+	//Optional
+	AuthorEmail  string `json:"author_email"`
+	Text         string `json:"text"`
+	CallDuration uint32 `json:"call_duration"`
+	CreatedAt    string `json:"created_at"`
+	UpdatedAt    string `json:"updated_at"`
+}
+
+// ListNoteParams = parameters for listing customer notes in API.
+type ListNotesParams struct {
+	CustomerUUID string `json:"customer_uuid"`
+	Cursor
+}
+
+// Notes is result of listing customer notes in API.
+type Notes struct {
+	Entries []*Note `json:"entries"`
+	Pagination
+}
+
+const (
+	singleCustomerNoteEndpoint = "customer_notes/:uuid"
```

Comment:
I don't like the naming but it seems to follow what already exists so ... :man_shrugging: